test/pylib/scylla_cluster.py

```diff
@@ -485,3 +487,41 @@ def after_test(self, name) -> None:
                                "the test must drop all keyspaces it creates.")
         for server in self.cluster:
             server.write_log_marker("------ Ending test {} ------\n".format(name))
+
+
+class ScyllaClusterManager:
+    """Manages a Scylla cluster for running test cases"""
+    cluster: ScyllaCluster
+
+    def __init__(self, test_name: str, clusters: Pool[ScyllaCluster]) -> None:
+        self.test_name: str = test_name
+        self.clusters: Pool[ScyllaCluster] = clusters
+
+    async def start(self) -> None:
+        """Start, start first cluster"""
+        await self._get_cluster()
+
+    async def stop(self) -> None:
+        """Stop, cycle last cluster if not dirty and present"""
+        self.cluster.after_test(self.test_name)
+        await self._return_cluster()
+
+    async def _get_cluster(self) -> None:
+        self.cluster = await self.clusters.get()
+        logging.info("Getting new Scylla cluster %s", self.cluster)
+
+    async def _return_cluster(self) -> None:
+        logging.info("Returning Scylla cluster %s", self.cluster)
+        await self.clusters.put(self.cluster)
+        del self.cluster
+
+
+@asynccontextmanager
+async def get_cluster_manager(test_name: str, clusters: Pool[ScyllaCluster]) \
+        -> AsyncIterator[ScyllaCluster]:
+    """Create a temporary manager for the active cluster used in a test
+       and provide the cluster to the caller."""
+    manager = ScyllaClusterManager(test_name, clusters)
+    await manager.start()
+    yield manager
+    await manager.stop()
```

Comment:
Don't we need to wrap this in `try...finally`? The examples in documentation do this:
https://docs.python.org/3/library/contextlib.html#contextlib.asynccontextmanager